### PR TITLE
[SPARK-42304][FOLLOWUP][SQL] Add test for `GET_TABLES_BY_TYPE_UNSUPPORTED_BY_HIVE_VERSION`

### PR DESCRIPTION
**Pull request by valentinp17**
### What changes were proposed in this pull request?
1) Add test for getTablesByType function to cover error class "UNSUPPORTED_FUNCTION_BY_HIVE_VERSION" added in PR https://github.com/apache/spark/pull/42706

### Why are the changes needed?
Test error class for getTablesByType()

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
./build/sbt "testOnly org.apache.spark.SparkThrowableSuite"
./build/sbt "sql/testOnly org.apache.spark.sql.hive.client.HiveShimSuites"

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by valentinp17** [comment]
@itholic @MaxGekk @cxzl25 PTAL when you find some time

File: .github/PULL_REQUEST_TEMPLATE [hunk 1]
```diff
@@ -9,7 +9,7 @@ Thanks for sending a pull request!  Here are some tips for you:
   7. If you want to add a new configuration, please read the guideline first for naming configurations in
      'core/src/main/scala/org/apache/spark/internal/config/ConfigEntry.scala'.
   8. If you want to add or modify an error type or message, please read the guideline first in
-     'core/src/main/resources/error/README.md'.
+     'common/utils/src/main/resources/error/README.md'.
```
Comment (itholic) [inline .github/PULL_REQUEST_TEMPLATE:12]: I think technically this should be in separate PR, but it's okay since it's a minor typo fix.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveShimSuite.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  test("getTablesByType") {
+    if (version >= "2.3") {
+      val managed = shim.getTablesByType(hive, "default", "table1", TableType.MANAGED_TABLE)
+      val external = shim.getTablesByType(hive, "default", "table1", TableType.EXTERNAL_TABLE)
+
+      assert(managed === Seq("table1"))
+      assert(external === Seq.empty)
+    } else {
+      val e = intercept[SparkUnsupportedOperationException] {
+        shim.getTablesByType(hive, "default", "table1", TableType.MANAGED_TABLE)
```
Comment (itholic) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveShimSuite.scala:70]: Maybe do we need to test for `EXTERNAL_TABLE` as well?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveShimSuite.scala [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    hiveConf.set("datanucleus.schema.autoCreateAll", "true")
+    hiveConf.set("hive.metastore.schema.verification", "false")
+    hiveConf.set("hive.metastore.warehouse.dir", warehousePath.toString)
+
+    getHive(hiveConf)
+  }
+
+  test("createTables") {
+    shim.createTable(hive, new Table("default", "table1"), ifNotExists = true)
```
Comment (itholic) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveShimSuite.scala:58]: qq: isn't there any chance to fail creating table here??

**Review by itholic — COMMENTED** [review]
Left some comments. Overall structure looks fine to me.

File: .github/PULL_REQUEST_TEMPLATE (hunk 1 above)
Comment (valentinp17) [reply .github/PULL_REQUEST_TEMPLATE]: Agree, I will create a new Jira ticket and PR for it

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveShimSuite.scala (hunk 2 above)
Comment (valentinp17) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveShimSuite.scala]: Added external table to test working versions and exception cases

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveShimSuite.scala (hunk 3 above)
Comment (valentinp17) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveShimSuite.scala]: I don't think so
I tried writing "HiveShimSuite" familiar to "HiveClientSuit", and there are no additional recovery plans for it either

**Comment by itholic** [comment]
Seems like the CI failure is not related to this change. Could you rebase to master?

**Comment by valentinp17** [comment]
@itholic Done!
But I mess up and made merge at first attempt. So I had to make force push rebase.

**Review by itholic — APPROVED** [review]
Looks pretty good to me. @MaxGekk could you happen to have an another look when you find some time?

**Comment by valentinp17** [comment]
@dongjoon-hyun could you please take a look and help decide whether these tests should be added or not.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveShimSuite.scala [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+class HiveShimSuite(version: String) extends HiveVersionSuite(version) {
+
+  private val shim = version match {
+    case "0.12" => new Shim_v0_12()
+    case "0.13" => new Shim_v0_13()
+    case "0.14" => new Shim_v0_14()
+    case "1.0" => new Shim_v1_0()
+    case "1.1" => new Shim_v1_1()
+    case "1.2" => new Shim_v1_2()
```
Comment (MaxGekk) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveShimSuite.scala:37]: This is removed by https://github.com/apache/spark/commit/3c84c229d167a6ab2857649e91fff6f0d57bb12c
cc @HyukjinKwon

Comment (HyukjinKwon) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveShimSuite.scala:37]: Yeah we don't support them anymore after JDK upgrade.

Comment (valentinp17) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveShimSuite.scala]: Done.
Deleted 0.12-1.2 versions and rebased to master.

**Review by itholic — APPROVED** [review]
LGTM.
